osidb/models.py

```diff
@@ -1050,6 +1055,43 @@ def _validate_ps_module_new_flaw(self):
                     f"for flaw with bz_id {bz_id}."
                 )
 
+    def _validate_sofware_collection(self):
+        """
+        Check that all RHSCL components in flaw's affects start with a valid collection.
+        """
+        if (
+            not self.ps_module.startswith("rhscl")
+            or self.ps_component in COMPONENTS_WITHOUT_COLLECTION
+        ):
+            return
+
+        streams = PsUpdateStream.objects.filter(ps_module__name=self.ps_module)
+        collections = streams.values_list("collections", flat=True).all()
+
+        is_valid_component = False
+        is_meta_package = False
+        for collection in collections:
+            for component in collection:
+                if self.ps_component.startswith(component + "-"):
+                    is_valid_component = True
+                    break
+                elif self.ps_component == component:
+                    is_meta_package = True
+                    break
+
+        if is_meta_package:
+            self.alert(
+                "invalid_software_collection_component",
+                f"PSComponent {self.ps_component} for {self.ps_module} indicates collection "
+                "meta-package rather than a specific component in the collection",
+            )
+        elif not is_valid_component:
+            self.alert(
+                "invalid_software_collection_component",
+                f"PSComponent {self.ps_component} for {self.ps_module} "
+                "does not match any valid collection",
+            )
```

Comment:
Is there a reason why originally two possible alerts which could occur at the same time (SFM2 behavior) this got transformed into only one possible alert? :thinking:

Reply:
oh, there is no real reason. It makes sense to be 2 separated alerts, I will submit a fix, thanks for pointing it out.